rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -128,6 +128,18 @@ class PublisherBase
   size_t
   get_intra_process_subscription_count() const;
 
+  /// Get the actual QoS settings, after the defaults have been determined.
+  /**
+   * The actual configuration applied when using RMW_QOS_POLICY_*_SYSTEM_DEFAULT
+   * can only be resolved after the creation of the publisher, and it
+   * depends on the underlying rmw implementation.
+   * If the underlying setting in use can't be represented in ROS terms,
+   * it will be set to RMW_QOS_POLICY_*_UNKNOWN.
+   * \return The actual qos settings. */
```

Comment:
Put the `*/` on its own line please.

Reply:
Also, please document in which cases this may throw an exception.

Reply:
Even if that's to say "may throw `runtime_error` when an unexpected error occurs" or something like that.